From eabe21164bbb29dead1a902b6d6e910f6892e3a7 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Fri, 7 Jun 2019 23:17:13 +0200
Subject: [PATCH] build-sys: fix static build gcrypt dependency gpg-error

The gcrypt depends on gpg-error, so find it's availability and link with it
when needed to fix static build.

Fixes:
- http://autobuild.buildroot.net/results/fb698e3e903869978bd5e69d791ec362317b7981

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Upstream status: commit eabe21164bbb29dead1a902b6d6e910f6892e3a7
---
 meson.build | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/meson.build b/meson.build
index 84e4360..3766edc 100644
--- a/meson.build
+++ b/meson.build
@@ -101,20 +101,22 @@ else
 endif

 crypto = get_option('USE_CRYPTO')
+crypto_dep = []
 if crypto == 'nettle'
-	crypto_dep = dependency('nettle')
+	crypto_dep += dependency('nettle')
	conf.set('USE_NETTLE', 1, description : 'If set use nettle crypto library.')
 elif crypto == 'gcrypt'
-	crypto_dep = cc.find_library('gcrypt')
+	crypto_dep += cc.find_library('gcrypt')
+	crypto_dep += dependency('gpg-error', required : true)
	conf.set('USE_GCRYPT', 1, description : 'If set use gcrypt crypto library.')
 elif crypto == 'openssl'
-	crypto_dep = dependency('openssl')
+	crypto_dep += dependency('openssl')
	conf.set('USE_OPENSSL', 1, description : 'if set use openssl crypto library.')
 elif crypto == 'kernel'
-	crypto_dep = dependency('disabler-appears-to-disable-executable-build', required : false)
+	crypto_dep += dependency('disabler-appears-to-disable-executable-build', required : false)
	conf.set('USE_KERNEL_CRYPTO_API', 1, description : 'if set use Linux kernel Crypto API.')
 elif crypto == 'none'
-	crypto_dep = dependency('disabler-appears-to-disable-executable-build', required : false)
+	crypto_dep += dependency('disabler-appears-to-disable-executable-build', required : false)
	conf.set('PING6_NONCE_MEMORY', 1,
		description : 'If set RFC6744 random does not use any CRYPTO lib.')
 endif
--
2.21.0
